### PR TITLE
CRUD Things, green tests

### PR DESCRIPTION
**Pull request by Fahrenhei7**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -22,19 +22,35 @@ class User < ApplicationRecord
...
+    case_sensitive: false
+  },
+  format: {
+    # this validation to avoid conflict between name and email of different
+    # users due signing in.
+    with: /^[a-zA-Z0-9_\.]*$/,
+    multiline: true
+  }
+
+  def associate(thing_to_associate)
```
Comment (Freika) [inline app/models/user.rb]: Rename argument to just `thing`

File: app/models/user.rb [hunk 2]
```diff
@@ -22,19 +22,35 @@ class User < ApplicationRecord
...
+    if thing_to_associate.is_a?(String)
+      thing = Thing.find_by_title(thing_to_associate)
```
Comment (Freika) [inline app/models/user.rb]: Use `find_by(title: ...)`

File: app/policies/application_policy.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    @user = user
+    @record = record
+  end
+
+  def index?
+    false
+  end
+
+  def show?
+    scope.where(:id => record.id).exists?
```
Comment (Freika) [inline app/policies/application_policy.rb]: https://github.com/arbox/ruby-style-guide/blob/master/README-ruRU.md#hash-literals

File: app/policies/thing_policy.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    @thing = thing
+  end
+
+
+  %w(edit? update? destroy?).each do |m|
+    define_method(m) do
+      user.things.include?(thing)
+    end
+  end
+
```
Comment (Freika) [inline app/policies/thing_policy.rb]: Remove unnecessary empty line.

File: spec/controllers/things_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,210 @@
+require 'rails_helper'
```
Comment (Freika) [inline spec/controllers/things_controller_spec.rb]: Use empty line to split contexts and tests from one another.

File: app/policies/application_policy.rb (hunk 3 above)
Comment (Fahrenhei7) [inline app/policies/application_policy.rb]: default pundit generation :(

**Pull request merged**